powerscale/provider/file_pool_policy_resource.go

```diff
@@ -0,0 +1,512 @@
+					int64validator.Between(1, 2147483647),
+				},
+			},
+			"description": schema.StringAttribute{
+				Description:         "A description for this File Pool Policy. (Update Supported)",
+				MarkdownDescription: "A description for this File Pool Policy. (Update Supported)",
+				Optional:            true,
+				Computed:            true,
+			},
+			"birth_cluster_id": schema.StringAttribute{
+				Description:         "The guid assigned to the cluster on which the policy was created.",
+				MarkdownDescription: "The guid assigned to the cluster on which the policy was created.",
+				Computed:            true,
+			},
+			"id": schema.StringAttribute{
+				Description:         "A unique name for this File Pool Policy.",
+				MarkdownDescription: "A unique name for this File Pool Policy.",
+				Computed:            true,
+			},
+			"state": schema.StringAttribute{
+				Description:         "Indicates whether this policy is in a good state (\"OK\") or disabled (\"disabled\").",
+				MarkdownDescription: "Indicates whether this policy is in a good state (\"OK\") or disabled (\"disabled\").",
+				Computed:            true,
+			},
+			"state_details": schema.StringAttribute{
+				Description:         "Gives further information to describe the state of this policy.",
+				MarkdownDescription: "Gives further information to describe the state of this policy.",
+				Computed:            true,
+			},
+		},
+	}
+}
+
+// Configure configures the resource.
+func (r *FilePoolPolicyResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	pscaleClient, ok := req.ProviderData.(*client.Client)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = pscaleClient
+}
+
+// Create allocates the resource.
+func (r *FilePoolPolicyResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	tflog.Info(ctx, "Creating File Pool Policy resource...")
+	var plan models.FilePoolPolicyModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	policyName := plan.Name.ValueString()
+	if err := helper.CreateFilePoolPolicy(ctx, r.client, &plan); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error creating File Pool Policy - %s", policyName),
+			err.Error(),
+		)
+		return
+	}
+
+	policyResponse, err := helper.GetFilePoolPolicy(ctx, r.client, plan.Name.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error getting File Pool Policy after creation",
+			err.Error(),
+		)
+		// if err, revert create
+		_ = helper.DeleteFilePoolPolicy(ctx, r.client, policyName)
```

Comment:
The delete operation may generate exception also and the error should not be ignored.

Reply:
this is revert action, if the creation is failed, we need to revert the creation. and the deletion is the revert action and thinking we should not let this err impact more.

Reply:
Just thinking the error need log down and let us know if `get` method or `delete` method fails, and since next step is `return`, it should not impact more

Reply:
@NavisJ the err is printed